packages/labs/eleventy-plugin-lit/README.md

```diff
@@ -1,68 +1,430 @@
+```js
+eleventyConfig.addWatchTarget('js/');
+```
 
-```html
-<script type="module">
-  // Hydrate template-shadowroots eagerly after rendering (for browsers without
-  // native declarative shadow roots)
-  import {
-    hasNativeDeclarativeShadowRoots,
-    hydrateShadowRoots,
-  } from './node_modules/@webcomponents/template-shadowroot/template-shadowroot.js';
-  if (!hasNativeDeclarativeShadowRoots) {
-    hydrateShadowRoots(document.body);
+## Usage
+
+Whenever you use a custom element in your Eleventy Markdown and HTML files,
+`@lit-labs/eleventy-plugin-lit` will automatically render its template and
+styles directly into your HTML.
+
+For example, given a markdown file `hello.md`:
+
+```md
+# Greetings
+
+<demo-greeter name="World"></demo-greeter>
+```
+
+And a component definition `js/demo-greeter.js`:
+
+<!-- prettier-ignore-start -->
+```js
+import {LitElement, html, css} from 'lit';
+
+class DemoGreeter extends LitElement {
+  static styles = css`
+    b { color: red; }
+  `;
+
+  static properties = {
+    name: {},
+  };
+
+  render() {
+    return html`Hello <b>${this.name}</b>!`;
   }
-  // ...
-  // Load and hydrate components lazily
-  import('./_js/components.bundle.js');
-</script>
+}
+customElements.define('demo-greeter', DemoGreeter);
+```
+<!-- prettier-ignore-end -->
+
+Then the Eleventy will produce `greeting/index.html`:
+
+<!-- prettier-ignore-start -->
+```html
+<h1>Greetings</h1>
+
+<demo-greeter name="World">
+  <template shadowroot="open">
+    <style>
+      b { color: red; }
+    </style>
+  </template>
+  Hello <b>World</b>!
+</demo-greeter>
 ```
+<!-- prettier-ignore-end -->
+
+The `<template shadowroot="open">` element above is an HTML standard called
+declarative shadow DOM. See the [Declarative Shadow
+DOM](#declarative-shadow-dom) section below for more details.
+
+### Component compatibility
+
+> 🚧 Note: Expanding this section with full details on component compatibility
+> is on the [roadmap](#roadmap). Follow
+> [#2494](https://github.com/lit/lit/issues/2494) for progress and discussion.
+> 🚧
+
+There are currently a number of restrictions that determine whether a component
+will be compatible with Lit pre-rendering, because not all of the component
+lifecycle methods are currently invoked, and the DOM APIs that can be used in
+certain lifecycle methods are restricted.
+
+The Lit team is working on finalizing and documenting the SSR lifecycle and
+restrictions, follow [#2494](https://github.com/lit/lit/issues/2494) for more
+details.
+
+### Passing data to components
+
+> 🚧 Note: Support for passing data as properties is on the [roadmap](#roadmap).
+> Follow [#2494](https://github.com/lit/lit/issues/2485) for progress and
+> discussion. 🚧
+
+Data can be passed to your components by setting attributes (see the `name`
+attribute in the [example](#usage) above), or by passing children into an
+element with a [`<slot>`](https://lit.dev/docs/components/shadow-dom/#slots).
```

Comment:
The element doesn't even need a `<slot>` to use children as data. It can iterate, query, etc. over children and interact with events.